From 172a11e2f3c86494c53ce9b0d4588b5bef3524e5 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Mon, 30 Jul 2018 21:36:11 +0200
Subject: [PATCH 6/9] sdk: Add migration for /missing/ custom button actions

* Since we removed onehand from settings
  we need to make sure user doesn't have them
  set in settings otherwise we will hit
  ArrayIndexOutOfBoundsException exception.
* Fixes : BUGBASH-1525

Change-Id: I7a5737ecfe5e70c3f0108c03b3e8745eade80433

# Conflicts:
#	packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
---
 .../lineagesettings/LineageDatabaseHelper.java        | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index 6ffe136..d88848d 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -324,6 +324,17 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
 
         if (upgradeVersion < 9) {
             if (mUserHandle == UserHandle.USER_OWNER) {
+                db.execSQL("UPDATE system SET value = '0' WHERE value IN ('10', '11') AND name IN ("
+                        + "'" + LineageSettings.System.KEY_HOME_LONG_PRESS_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_HOME_DOUBLE_TAP_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_MENU_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_MENU_LONG_PRESS_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_ASSIST_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_ASSIST_LONG_PRESS_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_APP_SWITCH_ACTION + "',"
+                        + "'" + LineageSettings.System.KEY_APP_SWITCH_LONG_PRESS_ACTION + "')");
+
+
                 db.beginTransaction();
                 SQLiteStatement stmt = null;
                 try {
-- 
2.17.1

